src/core/decorators/addClaimableRewards.ts

```diff
@@ -27,18 +27,25 @@ export function AddClaimableRewards({
 
           await Promise.all(
             protocolTokens.map(async (protocolToken) => {
-              const [reward] = await rewardAdapter.getPositions({
-                ...input,
-                protocolTokenAddresses: [protocolToken.address],
-              })
+              try {
+                const [reward] = await rewardAdapter.getPositions({
+                  ...input,
+                  protocolTokenAddresses: [protocolToken.address],
+                })
 
-              if (reward) {
-                if (reward.tokens) {
+                if (reward && reward.tokens && reward.tokens.length > 0) {
                   protocolToken.tokens = [
                     ...(protocolToken.tokens ?? []),
                     ...reward.tokens,
                   ]
                 }
+              } catch (error) {
+                if (
+                  (error as Error).message === 'Protocol token pool not found'
+                ) {
+                  return
+                }
```

Comment:
```suggestion
                if (error instaceOf Error && error.message === 'Protocol token pool not found') {
                  return
                }
```

If the error is not an instance of error for whatever reason and is just a string (it can happen), `error.message` would crash.

My worry is that error is not really standarised, I believe we only use it for the `SimplePoolAdapter`, not even for the `DefaultAdapter`, but for now this should do the trick.